Source/compositorclient/RPI/Implementation.cpp

```diff
@@ -183,6 +184,93 @@ class Platform {
 
 class Platform {
 private:
+    static constexpr uint16_t VIDEO_LAYER = 10000;
+    static constexpr uint16_t CURSOR_LAYER = VIDEO_LAYER + 1;
+    class Cursor {
+    public:
+        Cursor(uint32_t width, uint32_t height)
+            : _cursorHandle(0)
+            , _pointerResource(0)
+            , _position({ 0, 0 })
+            , _displaySize({ width, height })
+            , _cursorSize({ 16, 16 })
+        {
+            uint32_t imagePtr;
+            VC_RECT_T rect;
+            vc_dispmanx_rect_set(&rect, 0, 0, CursorData::width, CursorData::height);
+            _pointerResource = vc_dispmanx_resource_create(VC_IMAGE_RGBA32, CursorData::width, CursorData::height, &imagePtr);
+            vc_dispmanx_resource_write_data(_pointerResource,
+                                            VC_IMAGE_RGBA32,
+                                            CursorData::width*4,
+                                            const_cast<uint8_t*>(CursorData::image),
+                                            &rect);
+        }
+
+        ~Cursor()
+        {
+            DISPMANX_UPDATE_HANDLE_T updateHandle = vc_dispmanx_update_start(0);
+            vc_dispmanx_resource_delete(_pointerResource);
+            vc_dispmanx_element_remove(updateHandle, _cursorHandle);
+            vc_dispmanx_update_submit_sync(updateHandle);
+        }
+
+        void Show()
+        {
+            //Cleanup cursors on the screen if any!
+            Hide();
+            DISPMANX_DISPLAY_HANDLE_T displayHandle = vc_dispmanx_display_open(0);
+            DISPMANX_UPDATE_HANDLE_T updateHandle = vc_dispmanx_update_start(0);
+            VC_RECT_T srcRect, destRect;
+            static VC_DISPMANX_ALPHA_T alpha = {
+                static_cast<DISPMANX_FLAGS_ALPHA_T>(DISPMANX_FLAGS_ALPHA_FROM_SOURCE), 255, 0
+            };
+
+            vc_dispmanx_rect_set(&srcRect, 0, 0, CursorData::width << 16, CursorData::height << 16);
+            vc_dispmanx_rect_set(&destRect, _position.first, _position.second, _cursorSize.first, _cursorSize.second);
+            _cursorHandle = vc_dispmanx_element_add(updateHandle,
+                                                    displayHandle,
+                                                    CURSOR_LAYER,
+                                                    &destRect,
+                                                    _pointerResource,
+                                                    &srcRect,
+                                                    DISPMANX_PROTECTION_NONE,
+                                                    &alpha,
+                                                    nullptr,
+                                                    DISPMANX_NO_ROTATE);
+            vc_dispmanx_update_submit_sync(updateHandle);
+        }
+
+        void Hide()
```

Comment:
Can we do hide and show using vc_dispmanx_element_change_attributes with required opacity, instead add and removing ?

Reply:
This is from the header file of API:

//New function added to VCHI to change attributes, set_opacity does not work there.
VCHPRE_ int VCHPOST_ vc_dispmanx_element_change_attributes( DISPMANX_UPDATE_HANDLE_T update,

I did not try but it has this comment.

Reply:
Ok